src/Microsoft.Identity.Client/Mats/Mats.cs

```diff
@@ -29,60 +23,29 @@ internal class Mats : IMats
         private readonly bool _isScenarioUploadDisabled;
         private readonly object _lockObject = new object();
 
-        public static IMats CreateMats(
-            bool isTelemetryDisabled,
-            AudienceType audienceType, 
-            string appName, 
-            string appVer, 
-            string sessionId,
-            Action<IMatsTelemetryBatch> dispatchAction)
+        public static IMats CreateMats(IPlatformProxy platformProxy, IMatsConfig matsConfig)
         {
-            return CreateMatsWithScopesAndResources(
-                isTelemetryDisabled,
-                audienceType,
-                appName,
-                appVer,
-                sessionId,
-                dispatchAction,
-                null,
-                null);
-        }
-
-        public static IMats CreateMatsWithScopesAndResources(            
-            bool isTelemetryDisabled,
-            AudienceType audienceType,
-            string appName,
-            string appVer,
-            string sessionId,
-            Action<IMatsTelemetryBatch> dispatchAction,
-            IEnumerable<string> telemetryAllowedScopes,
-            IEnumerable<string> telemetryAllowedResources)
-        {
-            // TODO: replace with MSAL's existing PlatformProxyFactory...
-            var proxy = PlatformProxyFactory.CreatePlatformProxy();
-            string dpti = proxy.GetDpti();
-            string deviceNetworkState = proxy.GetDeviceNetworkState();
-            int osPlatformCode = proxy.GetOsPlatformCode();
+            string dpti = platformProxy.GetDpti();
+            string deviceNetworkState = platformProxy.GetDeviceNetworkState();
+            int osPlatformCode = platformProxy.GetMatsOsPlatformCode();
             
             bool enableAggregation = true;
             IEventFilter eventFilter = new EventFilter(enableAggregation);
-
             var errorStore = new ErrorStore();
-
             var scenarioStore = new ScenarioStore(TimeConstants.ScenarioTimeoutMilliseconds, errorStore);
 
             var allowedScopes = new HashSet<string>();
-            if (telemetryAllowedScopes != null)
+            if (matsConfig.AllowedScopes != null)
             {
-                foreach (string s in telemetryAllowedScopes)
+                foreach (string s in matsConfig.AllowedScopes)
                 {
                     allowedScopes.Add(s);
                 }
             }
             var allowedResources = new HashSet<string>();
-            if (telemetryAllowedResources != null)
+            if (matsConfig.AllowedResources != null)
             {
-                foreach (string s in telemetryAllowedResources)

```

Comment:
allowedScopes.UnionWith(matsConfig.AllowedScopes)

Although allowedScopes should maybe be an ISet in the first place? And the class MatsConfig should not allow nulls, default to empty collections instead.